frame/offences/src/migration.rs

```diff
@@ -41,8 +41,9 @@ pub fn remove_deferred_storage<T: Config>() -> Weight {
 	let mut weight = T::DbWeight::get().reads_writes(1, 1);
 	let deferred = <DeferredOffences<T>>::take();
```

Comment:
This field was removed hence why you don't see it being written anymore. The migration deals with the fact that it was removed. You are correct though, you should hardcode `DisableStrategy::WhenSlashed` here instead of trying to read it from storage.